### PR TITLE
Adding more debug logging for runners.py

### PR DESCRIPTION
**Pull request by max-rocket-internet**
It's almost impossible to investigate issues like https://github.com/locustio/locust/issues/1535 and some of my previous issues without making custom local changes to add more logging, which is annoying and time consuming.

So here I'm adding some more debug log messages and add a function for updating the `state`, with a debug log message.

### CODE REVIEW

**Comment by cyberw** [comment]
lgtm.

File: locust/runners.py [hunk 1]
```diff
@@ -578,27 +586,33 @@ def start(self, user_count, spawn_rate):
...
+            logger.debug("Sending spawn message to client %s" % (client.id))
             self.server.send_to_client(Message("spawn", data, client.id))
 
-        self.state = STATE_SPAWNING
+        self.update_state(STATE_SPAWNING)
 
     def stop(self):
         if self.state not in [STATE_INIT, STATE_STOPPED, STATE_STOPPING]:
-            self.state = STATE_STOPPING
+            logger.debug("Stopping...")
```
Comment (cyberw) [inline locust/runners.py:596]: oh, but I think this line is no longer needed. you can remove it as part of the actual fix later on :)

Comment (max-rocket-internet) [reply locust/runners.py:596]: Why is it not needed?

Comment (cyberw) [reply locust/runners.py:596]: Because we’re also going to be logging ”Updating state to STATE_CLEANUP, old state was X” right after that.

**Pull request merged**